sklearn/calibration.py

```diff
@@ -25,6 +27,337 @@
+        self.method = method
+        self.beta = beta
+        self.scoring = scoring
+        self.pos_label = pos_label
+        self.threshold = threshold
+
+    def fit(self, X, y):
+        """Select a decision threshold for the fitted model's positive class
+        using one of the available methods
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data
+
+        y : array-like, shape (n_samples,)
+            Target values
+
+        Returns
+        -------
+        self : object
+            Instance of self
+        """
+        y_score = _get_binary_score(self.base_estimator, X, self.scoring,
+                                    self.pos_label)
+        if self.method == 'f_beta':
+            precision, recall, thresholds = precision_recall_curve(
+                y, y_score, self.pos_label
+            )
+            f_beta = (1 + self.beta**2) * (precision * recall) /\
+                     (self.beta**2 * precision + recall)
+            self.decision_threshold_ = thresholds[np.argmax(f_beta)]
+            return self
+
+        fpr, tpr, thresholds = roc_curve(y, y_score, self.pos_label)
+
+        if self.method == 'roc':
+            # we find the threshold of the point (fpr, tpr) with the smallest
+            # euclidean distance from the "ideal" corner (0, 1)
+            self.decision_threshold_ = thresholds[
+                np.argmin(fpr ** 2 + (tpr - 1) ** 2)
+            ]
+        elif self.method == 'max_tpr':
+            indices = np.where(1 - fpr >= self.threshold)[0]
+            max_tpr_index = np.argmax(tpr[indices])
+            self.decision_threshold_ = thresholds[indices[max_tpr_index]]
+        else:
+            indices = np.where(tpr >= self.threshold)[0]
+            max_tnr_index = np.argmax(1 - fpr[indices])
+            self.decision_threshold_ = thresholds[indices[max_tnr_index]]
+        return self
+
+
+def _get_binary_score(clf, X, scoring=None, pos_label=1):
+    """Binary classification score for the positive label (0 or 1)
+
+    Returns the score that a binary classifier outputs for the positive label
+    acquired either from decision_function or predict_proba
+
+    Parameters
+    ----------
+    clf : object
+        Classifier object to be used for acquiring the scores. Needs to have
+        a decision_function or a predict_proba method
+
+    X : array-like, shape (n_samples, n_features)
+        The samples
+
+    pos_label : int, optional (default=1)
+        The positive label. Can either be 0 or 1
+
+    scoring : str or None, optional (default=None)
+        The method to be used for acquiring the score. Can either be
+        "decision_function" or "predict_proba" or None. If None then
+        decision_function will be used first and if not available
+        predict_proba
+    """
+    if not scoring:
+        try:
+            y_score = clf.decision_function(X)
+            if pos_label == 0:
```

Comment:
this should happen not if pos_label == 0, but if `pos_label == clf.classes_[0]` (and `len(clf.classes_) == 2`)